--- JsonReader.js       2007-04-01 22:05:00.000000000 +0200
+++ JsonReader.js.new   2007-04-12 02:10:51.000000000 +0200
@@ -40,8 +40,13 @@
                var id = (n[sid] !== undefined && n[sid] !== "" ? n[sid] : null);
                for(var j = 0, jlen = fields.length; j < jlen; j++){
                    var f = fields.items[j];
-                   var map = f.mapping !== undefined && f.mapping !== null ? f.mapping : f.name;
-                var v = n[map] !== undefined ? n[map] : f.defaultValue;
+                   if ((f.localroot !== undefined) && (n[f.localroot] !== undefined)) {
+                           var map = f.mapping !== undefined && f.mapping !== null ? f.mapping : f.name;
+                           var v = n[f.localroot][map] !== undefined ? n[f.localroot][map] : f.defaultValue;
+                   }else{
+                       var map = f.mapping !== undefined && f.mapping !== null ? f.mapping : f.name;
+                       var v = n[map] !== undefined ? n[map] : f.defaultValue;
+                   }
                    v = f.convert(v);
                    values[f.name] = v;
                }
